tests/check/check_h26xsigned_auth.c

```diff
@@ -1625,6 +1625,521 @@ START_TEST(vendor_axis_communications_operation)
+  ck_assert_int_eq(sv_rc, SV_NOT_SUPPORTED);
+
+  // Free nalu_list_item and session.
+  nalu_list_free_item(sei);
+  nalu_list_free_item(i_nalu);
+  signed_video_free(sv_camera);
+  signed_video_free(sv_vms);
+  free(private_key);
+
+  // Ownership of |nalu_to_prepend.nalu_data| has been transferred. Do not free memory
+
+}
+END_TEST
+
+/* Test description */
+START_TEST(public_key_on_validation_side_later)
+{
+  // This test runs in a loop with loop index _i, corresponding to struct sv_setting _i in
+  // |settings|; See signed_video_helpers.h.
+
+  SignedVideoReturnCode sv_rc;
+  SignedVideoCodec codec = settings[_i].codec;
+  sign_algo_t algo = settings[_i].algo;
+  SignedVideoAuthenticityLevel auth_level = settings[_i].auth_level;
+  signed_video_nalu_to_prepend_t nalu_to_prepend = {0};
+  nalu_list_item_t *i_nalu = nalu_list_item_create_and_set_id("I", 0, codec);
+  nalu_list_item_t *sei = NULL;
+  char *private_key = NULL;
+  size_t private_key_size = 0;
+
+  // On camera side
+  signed_video_t *sv_camera = signed_video_create(codec);
+  ck_assert(sv_camera);
+  // Read and set content of private_key.
+  sv_rc = signed_video_generate_private_key(algo, "./", &private_key, &private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_set_private_key(sv_camera, algo, private_key, private_key_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // No public key in SEI
+  sv_rc = signed_video_set_public_key_in_sei(sv_camera, false);
+  // Setting validation level.
+  sv_rc = signed_video_set_authenticity_level(sv_camera, auth_level);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
+  // Add an I-NALU to trigger a SEI.
+  sv_rc = signed_video_add_nalu_for_signing(sv_camera, i_nalu->data, i_nalu->data_size);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sv_rc = signed_video_get_nalu_to_prepend(sv_camera, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  sei = nalu_list_create_item(nalu_to_prepend.nalu_data, nalu_to_prepend.nalu_data_size, codec);
+  ck_assert(!tag_is_present(sei, codec, PUBLIC_KEY_TAG));
+
+  sv_rc = signed_video_get_nalu_to_prepend(sv_camera, &nalu_to_prepend);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(nalu_to_prepend.prepend_instruction == SIGNED_VIDEO_PREPEND_NOTHING);
+
+  // On validation side
+  signed_video_t *sv_vms = signed_video_create(codec);
+
+  // Validate this first GOP.
+  signed_video_authenticity_t *auth_report = NULL;
+
+  sv_rc = signed_video_add_nalu_and_authenticate(sv_vms, i_nalu->data, i_nalu->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+  ck_assert(!auth_report);
+  // Set public key
+  sv_rc = signed_video_set_public_key(sv_vms, sv_camera->signature_info->public_key, sv_camera->signature_info->public_key_size);
+  ck_assert_int_eq(sv_rc, SV_NOT_SUPPORTED);
+
+  sv_rc = signed_video_add_nalu_and_authenticate(sv_vms, i_nalu->data, i_nalu->data_size, &auth_report);
+  ck_assert_int_eq(sv_rc, SV_OK);
+
```

Comment:
You should check `auth_report` and also free the data, if present, to avoid memory leakage.